test/widgets/theme_test.dart

```diff
@@ -68,4 +75,40 @@ void main() {
     doCheck('TextButton',
       button: TextButton(onPressed: () {}, child: const Text(buttonText)));
   });
+
+  group('colorSwatchFor', () {
+    void doTest(int baseColor, Brightness brightness) {
+      final description = '${Color(baseColor).toString()}, $brightness';
+      testWidgets(description, (WidgetTester tester) async {
+        addTearDown(testBinding.reset);
+
+        tester.platformDispatcher.platformBrightnessTestValue = brightness;
+        addTearDown(tester.platformDispatcher.clearPlatformBrightnessTestValue);
+
+        final subscription = eg.subscription(eg.stream(), color: baseColor);
+
+        await tester.pumpWidget(const ZulipApp());
+        await tester.pump();
+
+        late StreamColorSwatch actualSwatch;
+        final navigator = await ZulipApp.navigator;
+        navigator.push(MaterialWidgetRoute(page: Builder(builder: (context) {
+          actualSwatch = colorSwatchFor(context, subscription);
+          return const Placeholder();
+        })));
+        await tester.pumpAndSettle();
+
+        final expectedSwatch = switch (brightness) {
+          Brightness.light => StreamColorSwatch.light(baseColor),
+          Brightness.dark  => StreamColorSwatch.dark(baseColor),
+        };
```

Comment:
Do you want to pass an `expected` parameter instead? I'm not sure how this `switch (brightness)` will handle the lerping case in the TODO below. (And the `dark` case is relevant only as part of implementing that TODO.)